pkg/ui/Flow.tsx

```diff
@@ -117,29 +116,57 @@ export class Flow<T extends Values> extends Component<Props<T>, State<T>> {
   }
 
   // Handles form submission
-  handleSubmit = (e: MouseEvent | FormEvent) => {
+  handleSubmit = (event: FormEvent<HTMLFormElement> | MouseEvent) => {
     // Prevent all native handlers
-    e.stopPropagation()
-    e.preventDefault()
+    event.stopPropagation()
+    event.preventDefault()
 
     // Prevent double submission!
     if (this.state.isLoading) {
       return Promise.resolve()
     }
 
+    const form = event.currentTarget
+
+    let body: T | undefined
+
+    if (form && form instanceof HTMLFormElement) {
+      const formData = new FormData(form)
+
+      // map the entire form data to JSON for the request body
+      body = Object.fromEntries(formData) as T
+
+      const hasSubmitter = (evt: any): evt is { submitter: HTMLInputElement } =>
+        "submitter" in evt
+
+      // We need the method specified from the name and value of the submit button.
+      // when multiple submit buttons are present, the clicked one's value is used.
+      if (hasSubmitter(event.nativeEvent)) {
+        const method = (
+          event.nativeEvent as unknown as { submitter: HTMLInputElement }
+        ).submitter
```

Comment:
You don't need this, when you use the type guard. That's the whole point :-)
```suggestion
        const method = event.nativeEvent.submitter
```

Reply:
ah lol, yeah. fixed